ui/components/app/confirm/info/row/row.tsx

```diff
@@ -54,61 +59,123 @@ const TOOLTIP_ICON_COLORS = {
+  severity?: Severity;
+}) {
+  if (!alertOwnerId) {
+    return null;
+  }
+
+  const { getFieldAlerts } = useAlerts(alertOwnerId);
+  const hasFieldAlert = getFieldAlerts(alertKey).length > 0;
+
+  if (!hasFieldAlert) {
+    return null;
+  }
+
+  return (
+    <>
+      <InlineAlert
+        onClick={() => {
+          // intentionally empty
+        }}
+        severity={severity}
+      />
+    </>
+  );
+}
+
 export const ConfirmInfoRow = ({
+  alertKey,
+  alertOwnerId,
   label,
   children,
   variant = ConfirmInfoRowVariant.Default,
   tooltip,
   style,
-}: ConfirmInfoRowProps) => (
-  <ConfirmInfoRowContext.Provider value={{ variant }}>
-    <Box
-      className="confirm-info-row"
-      display={Display.Flex}
-      flexDirection={FlexDirection.Row}
-      justifyContent={JustifyContent.spaceBetween}
-      flexWrap={FlexWrap.Wrap}
-      backgroundColor={BACKGROUND_COLORS[variant]}
-      borderRadius={BorderRadius.LG}
-      marginTop={2}
-      marginBottom={2}
-      paddingLeft={2}
-      paddingRight={2}
-      color={TEXT_COLORS[variant] as TextColor}
-      style={{
-        overflowWrap: OverflowWrap.Anywhere,
-        minHeight: '24px',
-        ...style,
-      }}
-    >
+}: ConfirmInfoRowProps) => {
+  const textColor = alertKey
+    ? getAlertTextColors(variant)
+    : TEXT_COLORS[variant];
+  return (
+    <ConfirmInfoRowContext.Provider value={{ variant }}>
       <Box
+        className="confirm-info-row"
         display={Display.Flex}
         flexDirection={FlexDirection.Row}
-        justifyContent={JustifyContent.center}
-        alignItems={AlignItems.center}
+        justifyContent={JustifyContent.spaceBetween}
+        flexWrap={FlexWrap.Wrap}
+        borderRadius={BorderRadius.LG}
+        backgroundColor={
+          alertKey
+            ? BACKGROUND_COLORS[ConfirmInfoRowVariant.Default]
+            : BACKGROUND_COLORS[variant]
+        }
+        marginTop={2}
+        marginBottom={2}
+        paddingLeft={2}
+        paddingRight={2}
+        color={textColor as TextColor}
+        style={{
+          overflowWrap: OverflowWrap.Anywhere,
+          minHeight: '24px',
+          ...style,
+        }}
       >
-        <Text variant={TextVariant.bodyMdMedium} color={TextColor.inherit}>
-          {label}
-        </Text>
-        {tooltip && tooltip.length > 0 && (
-          <Tooltip title={tooltip} style={{ display: 'flex' }}>
-            <Icon
-              name={TOOLTIP_ICONS[variant]}
-              marginLeft={1}
-              color={TOOLTIP_ICON_COLORS[variant] as unknown as IconColor}
-            />
-          </Tooltip>
-        )}
+        <Box
+          display={Display.Flex}
+          flexDirection={FlexDirection.Row}
+          justifyContent={JustifyContent.center}
+          alignItems={AlignItems.center}
+          gap={1}
+        >
+          <Text variant={TextVariant.bodyMdMedium} color={TextColor.inherit}>
+            {label}
+          </Text>
+          <RowAlert
```

Comment:
Following the discussion that this is used in snap UI, I wonder if an even safer and decoupled approach would be to render an abstract `children` property here and create a separate `AlertRow` component that renders a row containing the inline alert.

Similar to the multiple and single modal approach, it would decouple alerts from the base `Row` component entirely.

It would also mean if a confirmation wanted to support alerts, they could do it explicitly and keep it typesafe as we would insist on them providing the alert key and alert owner ID.

Reply:
I love the idea. I've created the `AlertRow` decoupled from `row.tsx`as you suggested.